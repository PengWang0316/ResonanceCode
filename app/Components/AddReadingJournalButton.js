import React from 'react';
import { Link } from 'react-router-dom';
import jQuery from 'jquery';

import styles from '../styles/AddReadingJournalButton.module.css';

const handleAddBtnClick = () => {
  const isShowAddBtns = jQuery('#addReadingBtn').css('opacity') !== '0';
  const opacity = isShowAddBtns ? '0' : '1';
  const readingBottom = isShowAddBtns ? '25px' : '85px';
  const journalBottom = isShowAddBtns ? '25px' : '145px';
  jQuery('#addReadingBtn').css({ opacity, bottom: readingBottom });
  jQuery('#addJournalBtn').css({ opacity, bottom: journalBottom });
};

// $ will use jQuery in the index.html.
const handleAddReadingButtonClick = () => $('#addReadingModal').modal('toggle');

const AddReadingJurnalButton = () => (
  <div role="button" tabIndex="-3" onClick={handleAddBtnClick}>
    {/* this $('#addReadingModal').modal('toggle') will use jQuery from index.html */}
    <div role="button" tabIndex="-4" onClick={handleAddReadingButtonClick} id="addReadingBtn" className={`${styles.addBtn} ${styles.addReadingBtn} text-center`}>Reading</div>
    <Link to={{ pathname: '/addJournal' }}><div id="addJournalBtn" className={`${styles.addBtn} ${styles.addJournalBtn} text-center`}>Journal</div></Link>
    <div className={`${styles.addBtn}`}><i className="fa fa-plus" /></div>
  </div>
);

export default AddReadingJurnalButton;
