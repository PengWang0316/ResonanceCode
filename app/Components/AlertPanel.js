import React from 'react';

import styles from '../styles/AlertPanel.module.css';

const AlertPanel = ({
  id = '', type = 'success', children = '', style = {}
}) => (
  <div id={id} className={`alert alert-${type} mt-3 ${styles.alertPanel}`} style={style} role="alert">
    {children}
  </div>
);
export default AlertPanel;
