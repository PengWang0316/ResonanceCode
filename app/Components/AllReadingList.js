import React from 'react';
import { connect } from 'react-redux';

import Pagination from './SharedComponents/Pagination';
import LoadingAnimation from './SharedComponents/LoadingAnimation';
import { fetchAllReadingList as fetchAllReading } from '../actions/ReadingActions';
import { NUMBER_OF_READING_PER_PAGE } from '../config';
import ReadingListRow from './ReadingListRow';

/* Show the all reading in the list */
export const AllReadingList = ({
  allReadingList, readingsAmount, fetchAllReadingList, handleClick, isLoading
}) => (
  <div className="modal fade" id="readingListModal" tabIndex="-1" role="dialog" aria-labelledby="readingListModalLabel" aria-hidden="true">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="readingListModalLabel">Select a reading for this journal</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <LoadingAnimation />
          {!isLoading &&
            <div className="mb-3">
              {allReadingList.map(reading => <ReadingListRow key={`${reading.name}_${reading.date}`} reading={reading} handleClick={handleClick} />)}
            </div>
          }

          {/* Pagination componentWillMount
            * Keeping readingsAmount, numberPerpage, and fetchContent out of AllReadingListPagination component can help it raising reusability.
          */}
          {readingsAmount !== null && <Pagination
            amount={readingsAmount}
            fetchContent={fetchAllReadingList}
            numberPerpage={NUMBER_OF_READING_PER_PAGE}
          />}

        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);
const mapStateToProps = state => ({
  allReadingList: state.allReadingList,
  readingsAmount: state.readingsAmount,
  isLoading: state.isLoading
});
const mapDispatchToProps = dispatch => ({
  fetchAllReadingList: pageNumber => dispatch(fetchAllReading(pageNumber))
});
// export AllReadingList;
export default connect(mapStateToProps, mapDispatchToProps)(AllReadingList);
