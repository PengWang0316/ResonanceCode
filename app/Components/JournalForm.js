import React, { Component } from "react";
import PropTypes from "prop-types";
import { matchDateFormat, getDateString, getCurrentDateString } from "../apis/Util";
import JournalContent from "./JournalContent";
import ReadingSearchAndList from "./ReadingSearchAndList";

class JournalForm extends Component {

  componentWillMount(){
    // Initial variables
    this.readings = this.props.journalData && this.props.journalData.pingPongStates ? this.props.journalData.pingPongStates : {}; // keep which readings should be attached on. Format is like {readingId: pingPongState}
    this.contents = {}; // Keep content keys and content
    this.contentKeyIndex = 0; // Generate keys for different contents
    this.contentIndexs = {}; // Use to track the index of content component in the array. Delete function needs it. The format is {contentKey: index}
    this.journalId = this.props.journalData ? this.props.journalData._id : null; // Keeping journal id for update
    this.userId = this.props.userId;

    this.state={
      journalDate: this.props.journalData ? getDateString(this.props.journalData.date) : getCurrentDateString(),
      isDateCorrect: true,
      isEmptyReading: this.props.journalData && this.props.journalData.pingPongStates ? false : true,
      contentComponentArray: [], // keep content component
      addJournalContent: "overview"
    };

    // console.log("Journal before alter: ", this.props.journalData);
    // If journal data exsit, get content
    if(this.props.journalData){
      // console.log(this.props.journalData.readingIds);
      // this.oldContentKeys = []; // keep original content keys
      this.readingIds = this.props.journalData.readingIds ? this.props.journalData.readingIds : null;
      this.oldReadingIds = this.readingIds ? Object.keys(this.props.journalData.readingIds) : null; // keep original reading ids

      // delete unnecessary properties from journal object
      let journal = Object.assign({}, this.props.journalData); /* Making a copy in order to prevent side effects */
      this.journalUserId = journal.user_id;
      delete journal.date;
      delete journal.user_id;
      // delete journal.ping_pong_state;
      delete journal._id;
      delete journal.readingIds;
      delete journal.pingPongStates;
      let keyRegExp = /-(\d+)$/; // The regular expression for subtract suffixs
      // console.log("Journal after alter: ", this.props.journalData);
      Object.keys(journal).map((key)=>{
        // this.oldContentKeys.push(key); // keeping the exsit keys in an array for delete function
        // handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared)
        // Also have to save the largest key number and use it to countine
        // journalData property is like {'overview-0': 'test overview', 'overview-0-isShared': true}
        // if key is isShared do not do anything
        let matchResult = key.match(keyRegExp);
        if (matchResult){
          if (this.contentKeyIndex < matchResult[1]) this.contentKeyIndex = matchResult[1]; // keep the max index number and user can contiune add new content
          this.handleAddContentClick(journal[key], key.replace(keyRegExp, ""), key, matchResult[1], journal[`${key}-isShared`]);
          // console.log("isShared: ",journal[`${key}-isShared`]);
        }
      });
      this.contentKeyIndex++;
    }

  }

  handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared){
    // put component in array in order to show
    newContentKey = newContentKey ? newContentKey : `${this.state.addJournalContent}-${this.contentKeyIndex}`;
    // put the component in the track object
    this.contentIndexs[newContentKey]=<JournalContent key={contentKeyIndex ? contentKeyIndex : this.contentKeyIndex++} newContent={addJournalContent ? addJournalContent : ""} newContentName={newContentName ? newContentName : this.state.addJournalContent} newContentKey={newContentKey} handleChangeCallback={(contentKey, contentText)=>{this.handleChangeCallback(contentKey, contentText);}} handleDeleteContentCallback={(contentKey)=>{this.handleDeleteContentCallback(contentKey);}} handleSharedBoxChangeCallback={(contentKey, isShared)=>{this.handleSharedBoxChangeCallback(contentKey, isShared)}} isShared={isShared ? true : false} />;
    this.setComponentToStateArray();
    // this.setState({contentComponentArray: this.state.contentComponentArray});

    // also have to put content title to state in order to track content and update
    this.contents[newContentKey] = addJournalContent ? addJournalContent : "";
    this.contents[`${newContentKey}-isShared`] = isShared ? true : false;
  }

  handleChangeCallback(contentKey, contentText){
    // console.log(contentKey, contentText);
    // update content in state
    this.contents[contentKey]=contentText;
  }

  handleDeleteContentCallback(contentKey){
    // Remove content from state
    if(this.contents.hasOwnProperty(contentKey)){
      delete this.contents[contentKey];
      delete this.contents[`${contentKey}-isShared`];
    }
    // Remove content component
    delete this.contentIndexs[contentKey];
    this.setComponentToStateArray();
    // this.state.contentComponentArray.splice(this.contentIndexs[contentKey],1);
    // this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  setComponentToStateArray(){
    // transfor object to array in order to display
    this.state.contentComponentArray=[];
    Object.keys(this.contentIndexs).map((element)=>{
      this.state.contentComponentArray.push(this.contentIndexs[element]);
    });
    this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  handleSharedBoxChangeCallback(contentKey, isShared){
    this.contents[`${contentKey}-isShared`] = isShared;
  }

  handleChange(event, element){
    let newState={};
    newState[element]=event.target.value;
    if(element=="journalDate") newState.isDateCorrect = matchDateFormat(newState[element]);
    this.setState(newState);
  }

  /* For ReadingSearchAndList callback */
  handleAttachReadingCallback(readingId){
    // console.log(this.readings);
    if(!this.readings[readingId]) this.readings[readingId] = "Neutral";
    this.setState({isEmptyReading: false});
  }

  handlePingpongstateChangeCallback(readingId, pingPongState){
    this.readings[readingId] = pingPongState;
    // console.log("pingPongState:", `${readingId} : ${pingPongState}`);
  }

  handleDetachAttachReadingCallback(readingId){
    delete this.readings[readingId];
    this.setState({isEmptyReading: Object.keys(this.readings).length===0});
  }

  handleCancel(){
    this.props.history.push("/reading");
  }

  handleDelete(event){
    // console.log("Delete journal!");
    event.preventDefault();
    this.props.handleDelete(this.journalId, Object.keys(this.readings), this.userId, this.oldReadingIds ? false : true);
    // console.log("**********delete************");
    // DatabaseApi.deleteJournal(this.journalId, Object.keys(this.readings), this.userId).then((reault)=>{this.props.history.push("/reading")});
  }

  handleSubmit(event){
    event.preventDefault();
    this.props.handleSubmit({
      isUpdate: this.props.journalData ? true : false,
      journalId: this.journalId,
      journalDate: this.state.journalDate,
      userId: this.userId,
      readings: this.readings,
      contents: this.contents,
      oldReadingIds: this.oldReadingIds
    });
  }

  render(){
      return(
        <div className="addReadingDiv">

          <div className="titleDiv">{this.props.journalData ? "Update journal for readings" : "Add a new journal for readings"}</div>
          <form className="form-horizontal" onSubmit={(event) => {this.handleSubmit(event);}}>

            <div className="text-right bottom-btn-div">
              {(!this.props.journalData || this.userId == this.journalUserId) &&
                <button type="submit" className="btn btn-info loginButton" disabled={this.props.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect)}>{this.props.journalData ? "Update" : "Submit"}</button>
              }
              {(this.props.journalData && this.userId == this.journalUserId) &&
                <button onClick={(event)=>{this.handleDelete(event)}} type="button" className="btn btn-danger loginButton">Delete</button>
              }
              <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
            </div>

              <div className="form-group row form-div">
                <div className="col-sm-6 row">
                  <div className="col-xs-3">
                    <label htmlFor="journalDate" className="col-sm-1 col-form-label">Date</label>
                  </div>
                  <div className="col-xs-9">
                    <input className={this.state.isDateCorrect?"form-control":"form-control form-control-warning"} type="text" placeholder="mm/dd/yyyy" id="journalDate" value={this.state.journalDate} onChange={(event)=>{this.handleChange(event, "journalDate")}} />
                    {!this.state.isDateCorrect && <span className="glyphicon glyphicon-warning-sign form-control-feedback form-control-warning-span"></span>}
                  </div>
                </div>
                <div className="col-sm-6">
                  (The date fromat is mm/dd/yyyy)
                </div>
              </div>



              {/*  Start reading search function */}
              <ReadingSearchAndList readings={this.props.journalData ? this.readingIds : null} pingPongStates={this.readings} attachReadingCallback={(readingId)=>{this.handleAttachReadingCallback(readingId);}} detachReadingCallback={(readingId)=>{this.handleDetachAttachReadingCallback(readingId);}} handlePingpongstateChangeCallback={(readingId, pingPongState)=>{this.handlePingpongstateChangeCallback(readingId, pingPongState);}}
              />


              {/* New content goes here */}
              {this.state.contentComponentArray}

              {/* Add content button and drop list */}
              <div className="row addJournalContentDiv">

                <div onClick={()=>{this.handleAddContentClick();}} className="addJournalContentBtnDiv col-sm-6"><i className="fa fa-plus-square" /> Add one content for your journal</div>
                <div className="col-sm-6">
                  <select  className="form-control" onChange={(event)=>{this.handleChange(event,"addJournalContent");}}>
                    <option value="overview">Overview</option>
                    <option value="internal_environment">Internal Environment</option>
                    <option value="relational_environment">Relational Environment</option>
                    <option value="physical_environment">Physical Environment</option>
                    <option value="creative_vector">Creative Vector</option>
                    <option value="synchronicities">Synchronicities</option>
                    <option value="dreams">Dreams</option>
                    <option value="dreams">Seasonal</option>
                    <option value="other">Other</option>
                  </select>

                </div>

              </div>


                <div className="text-right bottom-btn-div">
                  {(!this.props.journalData || this.userId == this.journalUserId) &&
                    <button type="submit" className="btn btn-info loginButton" disabled={this.props.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect)}>{this.props.journalData ? "Update" : "Submit"}</button>
                  }
                  {(this.props.journalData && this.userId == this.journalUserId) &&
                    <button onClick={(event)=>{this.handleDelete(event)}} type="button" className="btn btn-danger loginButton">Delete</button>
                  }
                  <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
                </div>
            </form>

          </div>
      );

  }

}
JournalForm.propTypes = {
  journalData: PropTypes.object,
  userId: PropTypes.string,
  handleSubmit: PropTypes.func.isRequired,
  isWriting: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired,
  handleDelete: PropTypes.func.isRequired
};
export default JournalForm;
