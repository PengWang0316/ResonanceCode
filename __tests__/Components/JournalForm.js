import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { JournalForm } from '../../app/Components/JournalForm';

// jest.mock('../../app/apis/Util', () => ({
//   matchDateFormat: jest.fn(),
//   getDateString: jest.fn(),
//   getCurrentDateString: jest.fn('01/02/2018')
// }));
jest.mock('../../app/Components/JournalContent', () => 'JournalContent');
jest.mock('../../app/Components/ReadingSearchAndList', () => 'ReadingSearchAndList');
jest.mock('../../app/Components/ImageUpload', () => 'ImageUpload');
jest.mock('../../app/actions/JournalActions', () => ({ deleteUploadImages: jest.fn() }));
jest.mock('jquery', () => jest.fn().mockReturnValue({ datepicker: jest.fn() }));
// window.jQuery = () => ({ datepicker: jest.fn() });
jest.mock('../../app/resources/jquery-ui.min', () => {});
describe('JournalForm test', () => {
  sinon.useFakeTimers(new Date(2018, 0, 2));
  const defaultProps = {
    journalData: {
      _id: 'journalId',
      date: new Date(2018, 0, 1),
      pingPongStates: null,
      readingIds: null,
      uploadImages: [],
      user_id: 'id',
      'overview-0': 'test overview',
      'overview-0-isPrivate': false,
      'overview-1': 'test overview 2',
      'overview-1-isPrivate': false,
      'other-2': 'other',
      'other-2-isPrivate': true
    },
    isWriting: false,
    history: { push: jest.fn() },
    handleSubmit: jest.fn(),
    handleDelete: jest.fn(),
    user: { _id: 'id' },
    clearJournalState: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<JournalForm {...props} />);

  test('intial states', () => {
    const component = getShallowComponent();
    expect(component.state('journalDate')).toBe('01/01/2018');
    expect(component.state('isDateCorrect')).toBeTruthy();
    expect(component.state('contentComponentArray').length).toBe(3);
    expect(component.state('addJournalContent')).toBe('overview_and_question');
    expect(component.state('uploadImages').length).toBe(0);
    expect(component.state('readings')).toEqual({});
    expect(component.state('readingIds')).toBeNull();

    const componentB = getShallowComponent({ ...defaultProps, journalData: null });
    expect(componentB.state('journalDate')).toBe('01/02/2018');

    const componentC = getShallowComponent({
      ...defaultProps,
      journalData: {
        _id: 'journalId',
        date: new Date(2018, 0, 1),
        pingPongStates: 'pingPongStates',
        readingIds: { id: 1 },
        uploadImages: [],
        user_id: 'id',
        'overview-0': 'test overview',
        'overview-0-isPrivate': false,
        'overview-1': 'test overview 2',
        'overview-1-isPrivate': false,
        'other-2': 'other',
        'other-2-isPrivate': true
      }
    });
    expect(componentC.state('readings')).toBe('pingPongStates');
    expect(componentC.state('readingIds')).toEqual({ id: 1 });
    expect(componentC.instance().oldReadingIds).toEqual(['id']);
  });

  test('componentDidMount', () => {
    getShallowComponent();
    const jquery = require('jquery');
    expect(jquery).toHaveBeenLastCalledWith('#journalDate');
    expect(jquery().datepicker).toHaveBeenCalled();
  });

  test('componentWillReceiveProps', () => {
    const component = getShallowComponent();
    expect(component.state('journalDate')).toBe('01/01/2018');
    component.setProps({
      journalData: {
        _id: 'newJournalId',
        date: new Date(2018, 1, 1),
        pingPonegStates: null,
        readingIds: null,
        uploadImages: [1, 2]
      }
    });
    expect(component.state('journalDate')).toBe('02/01/2018');
    // expect(component.state('uploadImages').length).toBe(2);
  });

  test('handleAddContentClick', () => {
    const component = getShallowComponent();
    expect(component.state('contentComponentArray').length).toBe(3);
    component.find({ role: 'button' }).simulate('click');
    expect(component.state('contentComponentArray').length).toBe(4);
    component.find({ role: 'button' }).simulate('click');
    expect(component.state('contentComponentArray').length).toBe(5);
  });

  test('handleChangeCallback', () => {
    const component = getShallowComponent();
    expect(component.instance().contents['overview-0']).toBe('test overview');
    expect(component.instance().contents['overview-1']).toBe('test overview 2');
    expect(component.instance().contents['other-2']).toBe('other');
    const journalContents = component.find('JournalContent');
    journalContents.at(0).prop('handleChangeCallback')('overview-0', 'new content A');
    journalContents.at(0).prop('handleChangeCallback')('overview-1', 'new content B');
    journalContents.at(0).prop('handleChangeCallback')('other-2', 'new content C');
    expect(component.instance().contents['overview-0']).toBe('new content A');
    expect(component.instance().contents['overview-1']).toBe('new content B');
    expect(component.instance().contents['other-2']).toBe('new content C');
  });

  test('handleDeleteContentCallback', () => {
    const component = getShallowComponent();
    expect(component.find('JournalContent').length).toBe(3);
    expect(component.instance().contents['overview-0']).toBe('test overview');
    expect(component.instance().contents['overview-0-isPrivate']).toBeFalsy();
    component.find('JournalContent').at(0).prop('handleDeleteContentCallback')('overview-0');
    expect(component.state('contentComponentArray').length).toBe(2);
    expect(component.instance().contents['overview-0']).toBeUndefined();
    expect(component.instance().contents['overview-0-isPrivate']).toBeUndefined();
  });

  test('handleChange', () => {
    const component = getShallowComponent();
    component.find('select').simulate('change', { target: { id: 'addJournalContent', value: 'thinking' } });
    expect(component.state('addJournalContent')).toBe('thinking');
    component.find('input').simulate('change', { target: { id: 'journalDate', value: 'not a date' } });
    expect(component.state('isDateCorrect')).toBeFalsy();
    component.find('input').simulate('change', { target: { id: 'journalDate', value: '01/01/2018' } });
    expect(component.state('isDateCorrect')).toBeTruthy();
    expect(component.state('journalDate')).toBe('01/01/2018');
  });

  test('handleKeyPress', () => {
    const component = getShallowComponent();
    const mockPreventDefaultA = jest.fn();
    component.find('input').simulate('keypress', { charCode: 1, preventDefault: mockPreventDefaultA });
    expect(mockPreventDefaultA).not.toHaveBeenCalled();
    const mockPreventDefaultB = jest.fn();
    component.find('input').simulate('keypress', { charCode: 13, preventDefault: mockPreventDefaultB });
    expect(mockPreventDefaultB).toHaveBeenCalled();
  });

  test('handleAttachReadingCallback', () => {
    const component = getShallowComponent();
    const readingId = 'newReadingId';
    expect(component.state('readings')[readingId]).toBeUndefined();
    component.find('ReadingSearchAndList').prop('attachReadingCallback')(readingId);
    expect(component.state('readings')[readingId]).toBe('Neutral');
  });

  test('handlePingpongstateChangeCallback', () => {
    const component = getShallowComponent();
    const readingId = 'newReadingId';
    expect(component.state('readings')[readingId]).toBeUndefined();
    component.find('ReadingSearchAndList').prop('handlePingpongstateChangeCallback')(readingId, 'newPingPongState');
    expect(component.state('readings')[readingId]).toBe('newPingPongState');
  });

  test('handleDetachAttachReadingCallback', () => {
    const component = getShallowComponent();
    const readingId = 'newReadingId';
    component.find('ReadingSearchAndList').prop('attachReadingCallback')(readingId);
    expect(component.state('readings')[readingId]).toBe('Neutral');
    component.find('ReadingSearchAndList').prop('detachReadingCallback')(readingId);
    expect(component.state('readings')[readingId]).toBeUndefined();
  });

  test('handleCancel', () => {
    const component = getShallowComponent();
    const actions = require('../../app/actions/JournalActions');
    component.instance().newImages = [1];
    component.find('button').at(2).simulate('click');
    expect(defaultProps.clearJournalState).toHaveBeenCalledTimes(1);
    expect(actions.deleteUploadImages).toHaveBeenCalledTimes(1);
    expect(defaultProps.history.push).toHaveBeenLastCalledWith('/reading');
  });

  test('handleDelete', () => {
    const component = getShallowComponent();
    const mockPreventDefault = jest.fn();
    const actions = require('../../app/actions/JournalActions');
    actions.deleteUploadImages = jest.fn();
    component.setState({ uploadImages: [{ id: 1 }] });
    component.find('button').at(1).simulate('click', { preventDefault: mockPreventDefault });
    expect(mockPreventDefault).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleDelete).toHaveBeenCalledTimes(1);
    expect(actions.deleteUploadImages).toHaveBeenCalledTimes(1);
  });

  test('handleSubmit', () => {
    const component = getShallowComponent();
    const mockPreventDefault = jest.fn();
    const actions = require('../../app/actions/JournalActions');
    actions.deleteUploadImages = jest.fn();
    component.instance().deleteImages = [1];
    component.find('form').simulate('submit', { preventDefault: mockPreventDefault });
    expect(mockPreventDefault).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleSubmit).toHaveBeenCalledTimes(1);
    expect(actions.deleteUploadImages).toHaveBeenCalledTimes(1);
  });

  test('handleImageDropCallback', () => {
    const component = getShallowComponent();
    component.setState({ uploadImages: [1] });
    component.instance().newImages = [{ id: 0 }];
    component.find('ImageUpload').prop('imageDropCallback')([{ id: 1 }, { id: 2 }]);
    expect(component.state('uploadImages').length).toBe(3);
    expect(component.instance().newImages.length).toBe(3);
  });

  test('handleDeleteImageCallback', () => {
    const component = getShallowComponent();
    component.setState({ uploadImages: [{ filePublicId: 0 }] });
    expect(component.instance().deleteImages.length).toBe(0);
    component.find('ImageUpload').prop('deleteImageCallback')('filePublicId');
    expect(component.instance().deleteImages[0]).toBe('filePublicId');
    expect(component.state('uploadImages').length).toBe(0);
  });

  test('handleSharedBoxChangeCallback', () => {
    const component = getShallowComponent();
    expect(component.instance().contents['overview-1-isPrivate']).toBeFalsy();
    expect(component.instance().contents['other-2-isPrivate']).toBeTruthy();
    component.find('JournalContent').at(0).prop('handleSharedBoxChangeCallback')('overview-1', true);
    component.find('JournalContent').at(0).prop('handleSharedBoxChangeCallback')('other-2', false);
    expect(component.instance().contents['overview-1-isPrivate']).toBeTruthy();
    expect(component.instance().contents['other-2-isPrivate']).toBeFalsy();
  });

  // test('No JournalData', () => {
  //   getShallowComponent({ ...defaultProps, journalData: null });
  // });

  test('JournalForm snapshot', () => expect(renderer.create(<JournalForm {...defaultProps} />).toJSON()).toMatchSnapshot());
});
