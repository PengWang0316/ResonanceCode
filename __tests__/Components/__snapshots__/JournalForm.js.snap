// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalForm test JournalForm snapshot 1`] = `
<div
  className="mainContentDiv"
>
  <div
    className="title"
  >
    Update journal for readings
  </div>
  <form
    className="form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="text-right mt-3"
    >
      <button
        className="btn btn-info submitButton"
        disabled={false}
        type="submit"
      >
        Update
      </button>
      <button
        className="btn btn-danger submitButton"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
      <button
        className="btn btn-secondary submitButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="form-group row mt-2"
    >
      <div
        className="col-sm-6 row"
      >
        <div
          className="col-xs-3"
        >
          <label
            className="col-sm-1 col-form-label"
            htmlFor="journalDate"
          >
            Date
          </label>
        </div>
        <div
          className="col-xs-9"
        >
          <input
            className="form-control"
            id="journalDate"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="mm/dd/yyyy"
            type="text"
            value="01/01/2018"
          />
        </div>
      </div>
      <div
        className="col-sm-6"
      >
        (The date fromat is mm/dd/yyyy)
      </div>
    </div>
    <ReadingSearchAndList
      attachReadingCallback={[Function]}
      detachReadingCallback={[Function]}
      existReadings={null}
      handlePingpongstateChangeCallback={[Function]}
      pingPongStates={Object {}}
    />
    <JournalContent
      handleChangeCallback={[Function]}
      handleDeleteContentCallback={[Function]}
      handleSharedBoxChangeCallback={[Function]}
      isPrivate={false}
      newContent="test overview"
      newContentKey="overview-0"
      newContentName="overview"
    />
    <JournalContent
      handleChangeCallback={[Function]}
      handleDeleteContentCallback={[Function]}
      handleSharedBoxChangeCallback={[Function]}
      isPrivate={false}
      newContent="test overview 2"
      newContentKey="overview-1"
      newContentName="overview"
    />
    <JournalContent
      handleChangeCallback={[Function]}
      handleDeleteContentCallback={[Function]}
      handleSharedBoxChangeCallback={[Function]}
      isPrivate={true}
      newContent="other"
      newContentKey="other-2"
      newContentName="other"
    />
    <div
      className="row addJournalContentDiv"
    >
      <div
        className="addJournalContentBtnDiv col-sm-6"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        <i
          className="fa fa-plus-square"
        />
         Add one content for your journal
      </div>
      <div
        className="col-sm-6"
      >
        <select
          className="form-control"
          id="addJournalContent"
          onChange={[Function]}
        >
          <option
            value="overview_and_question"
          >
            Overview and question
          </option>
          <option
            disabled={true}
            value="experience_of_i"
          >
            Experience of “I”
          </option>
          <option
            value="thinking"
          >
                Thinking
          </option>
          <option
            value="sensing"
          >
                Sensing
          </option>
          <option
            value="feeling"
          >
                Feeling
          </option>
          <option
            value="encoding"
          >
                Encoding
          </option>
          <option
            value="acting_creating"
          >
                Acting/creating
          </option>
          <option
            disabled={true}
            value="wxperience_of_the_fractal"
          >
             Experience of the Fractal Representative
          </option>
          <option
            value="personal_relation"
          >
                Personal relation
          </option>
          <option
            value="collective_event"
          >
                Collective event
          </option>
          <option
            value="abstract_entity"
          >
                Abstract entity
          </option>
          <option
            disabled={true}
            value="Subconscious"
          >
            Subconscious
          </option>
          <option
            value="physical_environment"
          >
                Physical environment
          </option>
          <option
            value="seasonal_marker"
          >
                Seasonal marker
          </option>
          <option
            value="synchronicity"
          >
                Synchronicity
          </option>
          <option
            value="dream"
          >
                Dream
          </option>
          <option
            value="highlights_from_i_ching"
          >
            Highlights from I-Ching
          </option>
          <option
            value="other"
          >
            Other
          </option>
        </select>
      </div>
    </div>
    <ImageUpload
      deleteImageCallback={[Function]}
      imageDropCallback={[Function]}
      uploadImages={Array []}
    />
    <div
      className="text-right mt-3"
    >
      <button
        className="btn btn-info submitButton"
        disabled={false}
        type="submit"
      >
        Update
      </button>
      <button
        className="btn btn-danger submitButton"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
      <button
        className="btn btn-secondary submitButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
